app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java

```diff
@@ -334,8 +334,11 @@ public void setCursor(final int index) {
         if (!hasFocus()) {
             requestFocus();
         }
-        post(() -> StringUtils.showSelection(this, index, index));
-        post(() ->setSelection(index));
+
+        post(() -> {
+            StringUtils.showSelection(this, index, index);
+            postDelayed(() -> setSelection(index), 200); // Run _after_ showSelection
```

Comment:
make sure this runs before autoformat is enabled again (on startup). otherwise, if you have a todo.txt file with a trailing line, or markdown with a 1./2./3. before last line, you get unintenionally something inserted.  
This possibly currently goes around that because selection is applied async later.

<https://github.com/gsantner/markor/commit/6014327f346af269533e7742397aad5fe52445d6>

Reply:
This shouldn't trigger any change in the text. It just shows the selection. I am pretty sure the bug with auto format and insertion was from the `setText`, not the `setSelection`. Will test for a few more days to confirm.

Also note that I added `withAutoFormatDisabled(final Callback.a0 callback)` in `HighlightingEditor`